tox.ini

```diff
@@ -0,0 +1,113 @@
+# Copyright 2024 Kmonas Sotirchos
+# See LICENSE file for licensing details.
+
+[tox]
+no_package = True
+skip_missing_interpreters = True
+env_list = format, lint, static, unit
+min_version = 4.0.0
+
+[vars]
+src_path = {tox_root}/src
+tests_path = {tox_root}/tests
+all_path = {[vars]src_path} {[vars]tests_path}
+
+[testenv]
+allowlist_externals = poetry, which
+deps=
+    poetry
+set_env =
+    PYTHONPATH = {tox_root}/lib:{[vars]src_path}
+    PYTHONBREAKPOINT=pdb.set_trace
+    PY_COLORS=1
+pass_env =
+    PYTHONPATH
+    CHARM_BUILD_DIR
+    MODEL_SETTINGS
+
+[testenv:update-requirements]
+description = Apply coding style standards to code
+deps =
+    poetry
+commands =
+    python --version
+    which python
+    poetry config warnings.export false
+    poetry lock
+    poetry install --no-root
+    poetry export -f requirements.txt -o requirements.txt --without-hashes
+
+[testenv:format]
+description = Apply coding style standards to code
+commands =
+    poetry install --with format --no-root
+    ruff format {[vars]all_path}
+    ruff check --fix {[vars]all_path}
+    poetry run black {[vars]all_path}
+
+[testenv:lint]
+description = Check code against coding style standards
+commands =
+    poetry install --with lint --no-root
+    # if this charm owns a lib, uncomment "lib_path" variable
+    # and uncomment the following line
+    # codespell {[vars]lib_path}
+    poetry run codespell {tox_root} \
+        --skip {tox_root}/poetry.lock \
+        --skip {tox_root}/venv \
+        --skip {tox_root}/.tox \
+        --skip {tox_root}/.git \
+        --skip {tox_root}/build
+    poetry run ruff check {[vars]all_path}
+    poetry run ruff format --check --diff {[vars]all_path}
```

Comment:
Maybe isort not since I see that it's not installed somewhere, but since we use `black` in the format check, I think it would make sense to run it during `lint` as well

Reply:
Yep, added it for lint checks as well

Reply:
1c0915b